src/api/resize/utils.js

```diff
@@ -0,0 +1,58 @@
+import viewport from 'viewport-list';
+import promisify from '../../utils/promisify';
+import { MESSAGES, getText } from '../../messages';
+
+
+const SIZE_RE = /(\d+)x(\d+)/;
+
+
+var getDevices = promisify(viewport);
+
+function parseSize (sizeDescription) {
+    var match = SIZE_RE.exec(sizeDescription);
+
+    if (!match)
+        return null;
+
+    var portraitWidth  = Number(match[1]);
+    var landscapeWidth = Number(match[2]);
+
+    if (Number.isNaN(portraitWidth) || Number.isNaN(landscapeWidth))
+        return null;
+
+    return { portraitWidth, landscapeWidth };
+}
+
+async function getDeviceSize (deviceName) {
+    var devices = await getDevices([deviceName]);
+
+    var size = null;
+
+    while (devices.length && !size)
+        size = parseSize(devices.shift().size);
+
+    return size;
+}
+
+export async function isDeviceSupported (device) {
+    var size = await getDeviceSize(device);
```

Comment:
Oops, we can't use async functions in validations. We need to figure out other way to do this